gammapy/utils/fitting/fit.py

```diff
@@ -207,59 +172,130 @@ def covar(self, backend="minuit"):
         result : `CovarResult`
             Results
         """
-        covar_backend = BackendRegistry.get("covar", backend)
+        compute = registry.get("covar", backend)
         parameters = self._model.parameters
 
         # TODO: wrap MINUIT in a stateless backend
         if backend == "minuit":
             if hasattr(self, "minuit"):
-                covariance_factors = covar_backend(self.minuit)
+                covariance_factors = compute(self.minuit)
             else:
-                raise RuntimeError("To use covar with minuit, you must first optimize.")
+                raise RuntimeError("To use minuit, you must first optimize.")
         else:
             function = self.total_stat
-            covariance_factors = covar_backend(parameters, function)
+            covariance_factors = compute(parameters, function)
 
         parameters.set_covariance_factors(covariance_factors)
 
         # TODO: decide what to return, and fill the info correctly!
         return CovarResult(model=self._model.copy(), success=True, nfev=0)
 
-    def run(self, optimize_opts=None, covar_opts=None):
+    def confidence(self, parameter, backend="minuit", sigma=1, maxcall=0):
+        """Estimate confidence interval.
+
```

Comment:
Add section `Parameters` to the docstring?

Reply:
Done